web/src/app/modules/views/main/editors/modules/tool-pallette/tools/common/select-tool.ts

```diff
@@ -82,84 +85,66 @@ export class SelectTool extends ToolBase implements IKeyboardTool, IDragAndDropT
         if (ctrl && evt.key === 'c') {
             // Copy
             this.cancelEvent(evt);
-            this.copySelection();
+            return this.copySelection();
         }
 
         if (ctrl && evt.key === 'v') {
             // Paste
             this.cancelEvent(evt);
-            this.pasteSelection();
+            return this.pasteSelection();
         }
 
         if (ctrl && evt.key === 'x') {
             // Cut
             this.cancelEvent(evt);
-            this.cutSelection();
+            return this.cutSelection();
         }
 
         if (evt.keyCode === Key.BACKSPACE) {
             // Delete
             this.cancelEvent(evt);
-            this.deleteSelection(this.selectedElementService.selectedElements.slice());
+            return this.deleteSelection();
         }
         return Promise.resolve();
     }
 
-    private static origin: IContainer;
-    private static cutFlag: boolean;
-    private static selection: IContainer[];
-
-    private copySelection(): void {
-        SelectTool.origin = this.model;
-        SelectTool.selection = this.selectedElementService.selectedElements.slice();
-        SelectTool.cutFlag = false;
+    private async copySelection(): Promise<void> {
+        // Copy Selection
+        SelectTool.originType = this.model.className;
```

Comment:
There should be a clipboard service. The select tool feels somehow wrong for the clipboard.